events/emitter.go

```diff
@@ -0,0 +1,62 @@
+package events
+
+import (
+	"context"
+	"sync"
+
+	"github.com/containerd/containerd/api/types/event"
+	"github.com/containerd/containerd/namespaces"
+	goevents "github.com/docker/go-events"
+)
+
+const (
+	EventVersion = "v1"
+)
+
+type Emitter struct {
+	sinks       map[string]*eventSink
+	broadcaster *goevents.Broadcaster
+	m           *sync.Mutex
```

Comment:
I updated to acquire before testing and holds when creating a new broadcaster.  If we don't need any locking for the broadcaster I believe that's the only thing that is mutated outside of the broadcaster so we could probably just remove it.  wdyt?